controllers/akodeploymentconfig/akodeploymentconfig_controller_intg_test.go

```diff
@@ -618,50 +647,52 @@ func intgTestAkoDeploymentConfigController() {
 					When("the cluster is being deleted ", func() {
 						When("the cluster is ready", func() {
 							BeforeEach(func() {
-								latestCluster := &clusterv1.Cluster{}
-								err := getCluster(latestCluster, cluster.Name, cluster.Namespace)
-								Expect(err).To(BeNil())
-								conditions.MarkTrue(latestCluster, akoov1alpha1.AviResourceCleanupSucceededCondition)
-								err = ctx.Client.Status().Update(ctx, latestCluster)
-								Expect(err).To(BeNil())
-								deleteObjects(latestCluster)
-
-								ensureRuntimeObjectMatchExpectation(client.ObjectKey{
-									Name:      cluster.Name,
-									Namespace: cluster.Namespace,
-								}, &clusterv1.Cluster{}, false)
+								deleteObjects(cluster)
 							})
 
 							//Reconcile -> reconcileNormal -> r.userReconciler.ReconcileAviUserDelete
 							It("should delete Avi user", func() {
-								ensureRuntimeObjectMatchExpectation(client.ObjectKey{
-									Name:      cluster.Name + "-" + "avi-credentials",
-									Namespace: cluster.Namespace,
-								}, &corev1.Secret{}, false)
+								secret := &corev1.Secret{
+									ObjectMeta: metav1.ObjectMeta{
+										Name:      cluster.Name + "-" + "avi-credentials",
+										Namespace: cluster.Namespace,
+									},
+								}
+								Eventually(func() bool {
+									return ensureObjectsDeleted(secret)
+								}, "60s", "5s").Should(BeTrue())
 							})
 						})
 						When("the cluster is not ready", func() {
 							BeforeEach(func() {
-								obj := &clusterv1.Cluster{}
+								obj := &clusterv1.Cluster{
+									ObjectMeta: metav1.ObjectMeta{
+										Name:      cluster.Name,
+										Namespace: cluster.Namespace,
+									},
+								}
 								err := getCluster(obj, cluster.Name, cluster.Namespace)
 								Expect(err).To(BeNil())
 								conditions.MarkFalse(obj, clusterv1.ReadyCondition, clusterv1.DeletingReason, clusterv1.ConditionSeverityInfo, "")
 								conditions.MarkTrue(obj, akoov1alpha1.AviResourceCleanupSucceededCondition)
-								err = ctx.Client.Status().Update(ctx, obj)
-								Expect(err).To(BeNil())
+								Eventually(func() error {
+									err = ctx.Client.Status().Update(ctx, obj)
```

Comment:
not very familiar with these tests but only nit i can think of that this could need a get then update in the Eventually? not sure if the Cluster resource could be modified by something else in parallel

Reply:
Make sense. And I remember this is not necessary when I used the newer version of kubebuilder. Let me remove it.